### PR TITLE
Retry only failed tests

### PR DESCRIPTION
**Pull request by usamasaqib**
### What does this PR do?
This PR adds support to only retry failed tests with the kernel matrix testing system.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: test/new-e2e/system-probe/test-json-review/main.go [hunk 1]
```diff
@@ -98,18 +61,38 @@ func reviewTests(jsonFile string) (string, error) {
 	}
```
Comment (val06) [inline test/new-e2e/system-probe/test-json-review/main.go:62]: `defer jf.Close()`

**Review by val06 — CHANGES_REQUESTED** [review]
cred

File: test/new-e2e/system-probe/test-json-review/main.go [hunk 2]
```diff
@@ -9,71 +9,30 @@ import (
...
-		if len(failedTests) > 0 {
-			fmt.Println(color.RedString(failedTests))
-		} else {
-			fmt.Println(color.GreenString(fmt.Sprintf("All tests cleared in attempt %d", i+1)))
-			return
-		}
+	if failedTests != "" {
+		fmt.Println(color.RedString(failedTests))
+	} else {
+		fmt.Println(color.GreenString("All test passed."))
```
Comment (brycekahle) [inline test/new-e2e/system-probe/test-json-review/main.go:34]:
```suggestion
		fmt.Println(color.GreenString("All tests passed."))
```

File: test/new-e2e/system-probe/test-json-review/main.go [hunk 3]
```diff
@@ -90,26 +49,51 @@ type testEvent struct {
...
+			continue
+		}
+		if res, ok := testResults[testKey(ev.Test, ev.Package)]; ok {
+			// If the test is already recorded as passed, it means the test
+			// eventually succeeded.
+			if res.Action == "pass" {
+				continue
+			}
+			if res.Action == "fail" && ev.Action == "pass" {
+				testResults[testKey(ev.Test, ev.Package)] = ev
```
Comment (brycekahle) [inline test/new-e2e/system-probe/test-json-review/main.go:82]: Do we want to record/output flaky tests separately?

Comment (usamasaqib) [reply test/new-e2e/system-probe/test-json-review/main.go:85]: They will be made available to the CI visibility backend. Not sure how useful reporting them in the job output is...

File: test/new-e2e/scenarios/system-probe/main.go [hunk 4]
```diff
@@ -3,6 +3,9 @@
 // This product includes software developed at Datadog (https://www.datadoghq.com/).
 // Copyright 2016-present Datadog, Inc.
 
+//go:build !windows
+// +build !windows
```
Comment (brycekahle) [inline test/new-e2e/scenarios/system-probe/main.go:7]: FYI you don't need the old format anymore

**Review by brycekahle — APPROVED** [review]

File: test/new-e2e/scenarios/system-probe/main.go (hunk 4 above)
Comment (usamasaqib) [reply test/new-e2e/scenarios/system-probe/main.go:7]: Which one is old?

**Review by val06 — APPROVED** [review]

**Pull request merged**